### ISSUE TITLE
Update "Installing the Knative CLI" on knative.dev & in README

### ISSUE THREAD

**Issue by rhuss** [knative/client#648]
## Describe the feature:

The [kn installation instruction] for the kn client is outdated and not well formatted:

* Missing introduction where to download released versions
* Describe how to make it executable and put it into the path
* Missing link to release notes
* Missing instruction how to run kn from Docker images (including volume mount to mount `~/.kubec/config`)
* Prerequisites for installing via go (i.e. Go 1.13)
* Should be included in the GitHub README of the client, too.
* Add a sample command (`kn version`) to verify whether kn is working properly

This story is part of #510 .

**Comment by rhuss** [comment]
The fiel to be updated is over there in the docs repo (where also the PR needs to be raised): https://github.com/knative/docs/blob/master/docs/install/install-kn.md

**Comment by MIBc** [comment]
I can have a try.

**Comment by rhuss** [comment]
Cool, thanks ! Actually it would be best to start over there at docs, to improve the installation documentation. Also, please track #400 as we are about to enable a brew based installation.

The README rework is long overdue and is the next on my task list. As soon as I'm there we should update the instructions here, too.

/assign MiBc

**Comment by kristianmandrup** [comment]
I've been unable to install the client on my MacOSX. Have been trying many different approaches as outlined in [315](https://github.com/knative/client/issues/315#issuecomment-588901218)

Hope you will soon have the `brew` based solution available. Cheers!

**Comment by rhuss** [comment]
@kristianmandrup what are the concrete issue, maybe we can fix it quickly. You can download the latest macOS client from https://github.com/knative/client/releases/download/v0.12.0/kn-darwin-amd64 . Then I'd recommend:

```
$ mv kn-darwin-amd64 kn
$ chmod a+x kn
$ ./kn version
```

If you a permission (or not-signer-error) because you are running on Mojave, then you can do the following:

* Open in Finder the directory with the downloaded binary
* Right-click on the binary and Choose "Open". 
* You get a warning, but you should accept to run the binary.
* This will flicker shortly in a window and then close.

After that, you can run the binary directly from the terminal as usual.
Apologies for this we are still on sorting out how to best sign the binary so that we don't need that workaround anymore.

**Comment by kristianmandrup** [comment]
Great. I have not seen these instruction on the install page of the Knative docs site.
Worked like a charm. When I tried this before, I had only used `+x` for `chmod`

```sh
kn version
Version:      v0.12.0
Build Date:   2020-01-29 21:03:45
Git Revision: 164cb5f
Supported APIs:
* Serving
  - serving.knative.dev/v1alpha1 (knative-serving v0.12.0)
* Eventing
  - sources.eventing.knative.dev/v1alpha1 (knative-eventing v0.12.0)
  - eventing.knative.dev/v1alpha1 (knative-eventing v0.12.0)
➜  Downloads kn service list
No services found.
```

**Comment by rhuss** [comment]
Sorry for that, sorry. We are currently are on it (e.g. this issue ;-)

**Comment by rhuss** [comment]
fixed with knative/docs#2230 

Maybe we can move that installation section to the user guide sugested in #650
